app/controllers/my-projects/assignment/recommendations/add.js

```diff
@@ -100,13 +101,18 @@ export default class MyProjectsProjectRecommendationsAddController extends Contr
 
   minDate = MINIMUM_VOTE_DATE;
 
+  // currently only used to name the universal queue
   queueName = 'recommendation';
 
   submitError = false;
 
-  @computed('fileQueue', 'queueName')
-  get recommendationAddQueue() {
-    return this.fileQueue.find(this.queueName);
+  @computed('fileQueue', 'dispositions')
+  get queuesByDisposition() {
+    const queuesByDisposition = {};
+    this.dispositions.forEach((disposition) => {
+      queuesByDisposition[disposition.id] = this.fileQueue.create(disposition.id);
+    });
+    return queuesByDisposition;
```

Comment:
You could alternatively return this as a `.reduce` off of `this.dispositions`

Reply:
Pro suggestion, thanks!

Reply:
updated!